models/src/main/scala/coop/rchain/models/rholang/sorter/ParSortMatcher.scala

```diff
@@ -29,8 +29,9 @@ private[sorter] object ParSortMatcher extends Sortable[Par] {
         locallyFree = par.locallyFree,
         connectiveUsed = par.connectiveUsed
       )
+      connectiveUsedScore = if (par.connectiveUsed) Score.CONNECTIVE_USED else 0
       parScore = Node(
-        Score.PAR,
+        Score.PAR + connectiveUsedScore,
```

Comment:
Scores for different types / fields should never be added as ints. Instead, they should be appended to the score tree.

ConnectiveUsed flag should be handled analogously to `persistentScore` in ReceiveSortMatcher, with one difference: I'd like the score tree leaf for this field be rightmost instead of leftmost in its parent node.

Reply:
What's the difference to `persistentScore` that makes you want it to be rightmost?